README.md

```diff
@@ -288,6 +288,37 @@ public class FailoverSampleApp2 {
   }
 }
 ```
+
+### Plugin Manager
+
+<div style="text-align:center"><img src="./docs/files/images/plugin_manager_diagram.png" /></div>
+
+Figure shows a simplified workflow of the plugin manager. Starting at the top, when a new query is executed by the application, it is passed into the plugin manager. From the  plugin manager, the query is passed in order of plugins loaded, in this figure, `Custom Plugin A` to `Custom Plugin B` and finally to `Default Plugin` which executes the query and returns the result sets back up the chain.
+Custom plugins can be used to interact with queries, such as adding a feature to monitor a connection's status in the case of `NodeMonitoringConnectionPlugin`. 
+
+By default, `NodeMonitoringConnectionPlugin` is loaded for [Enhanced Failure Monitoring](https://github.com/awslabs/aws-mysql-jdbc#aws-iam-database-authentication). Additional custom plugins can be included and chained together.
+
+To learn how to write custom plugins, refer to examples located inside [Custom Plugins Demo](https://github.com/awslabs/aws-mysql-jdbc/tree/main/src/demo/java/demo/customplugins).
+
+**Note**: The order of which plugins load in matters.
+
+#### Plugin Manager Parameters
+| Parameter       | Value           | Required      | Description  |
+| ------------- |:-------------:|:-------------:| ----- |
+|`connectionPluginFactories` | String | No | String of custom plugins to use monitoring. If left unset, `NodeMonitoringConnectionPluginFactory` will be loaded. <br/><br/>Each factory in the string should be separated by `,`<br/><br/>NOTE: The order of factories declared matters.  <br/><br/>Example `MyPluginA`, `MyPluginA,MyPluginB,NodeMonitoringConnectionPluginFactory` <br/><br/>**Default value:** `NodeMonitoringConnectionPluginFactory` |
+
+### Enhanced Failure Monitoring
+Enhanced failure monitoring, part of `NodeMonitoringConnectionPlugin`, periodically pings the database node to check if the database node is still healthy/available to be used. In the case of the database node showing up as downed/unhealthy, `NodeMonitoringConnectionPlugin` will automatically failover to a different database node.   
+
+#### Enhanced Failure Monitoring Parameters
+| Parameter       | Value           | Required      | Description  |
+| ------------- |:-------------:|:-------------:| ----- |
+|`nativeFailureDetectionEnabled` | Boolean | No | Set to false if you would like to disable enhanced failure monitoring feature. <br/><br/>**Default value:** `true` |
```

Comment:
Need to rename "nativeFailureDetectionEnabled" to be consistent with other parameters. Let's say "failureDetectionEnabled"